src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1514,13 +1519,9 @@ theorem msb_ushiftRight {x : BitVec w} {n : Nat} :
   case succ nn ih =>
     simp [BitVec.ushiftRight_eq, getMsbD_ushiftRight, BitVec.msb, ih, show nn + 1 > 0 by omega]
 
-/-! ### ushiftRight reductions from BitVec to Nat -/
-
 @[simp]
```

Comment:
This lemma must not be tagged as simp. As you can see from `ushiftRight_eq'` the simplifier already rewrite `x >>> y` to `x >>> y.toNat` if `y : BitVec w` so this specific lemma is not going to work properly with the default simp set in its current form.

We should instead provide
```
n >>> n.toNat = 0
```
for the default `simp` set. As almost nothing in the library is currently talking about `x >>> y` for `y : BitVec w` (see [loogle query](https://loogle.lean-lang.org/?q=HShiftRight.hShiftRight%2C+BitVec)] I would suggest to instead only provide `n >>> n.toNat = 0` in `Init.Data.BitVec.Lemmas` and prove `n >>> n = 0` in `BVDecide.Normalize` from it (after all this theorem should follow from `simp` with `x >>> y = x >>> y.toNat` and `x >>> x.toNat = 0`)

Reply:
Ah yes, makes sense. Thanks for pointing it out.